chain-impl-mockchain/src/accounting/account/account_state.rs

```diff
@@ -84,6 +84,8 @@ pub struct AccountState<Extra> {
     pub value: Value,
     pub tokens: Hamt<DefaultHasher, TokenIdentifier, Value>,
     pub last_rewards: LastRewards,
+    #[cfg(feature = "evm")]
+    pub evm_state: chain_evm::state::AccountState,
     pub extra: Extra,
```

Comment:
It is also an option, but I would prefer to add a new field because it does not leads to the huge update related for replacing in all parts usage of the `Account<chain_evm::state::AccountState>` with the `Account<()>` as it is used currently.
And generally I want to suggest to remove this optional `extra` field, as it is not used and defined as `()` type.

Reply:
> Would it be easier to treat all those states as the Extra instead of adding an hardcoded field for each one?

I am not sure that it will be easier, because still we will have a separate struct with which we will define this `Extra` generic type. But again we will be need to correctly update AccountState typing in all places where it is initialized, that is does not look like as an easy update.
Instead of adding just a new field and modify a few places.

Reply:
> AccountState typing in all places where it is initialized

I would expect that to be just a type definition in one place and then compiler inference all the way down, but I may be wrong. 
To be clear, I don't think adding a new field is a wrong solution, I'm just a bit worried of bloating the struct with many fields, and I'm wondering why do we even have that extra field if we don't use it.

Reply:
Actually I am not sure, what was an intention of the `extra`, so I am also wondering what is the purpose of this field :)

Reply:
I am also wondering what is the purpose of the `extra` field for the `AccountState` 😊